### DISCUSSION TITLE
Why not use git submodule to manage the third dependence?

### DISCUSSION THREAD

**Discussion by peerless2012** [Tencent/tgfx#16] [Q&A] [answered]
Use git submodule is more friendly to Windows platform, use `git submodule update --init --recursive` and do not need to install npm or others.

**Comment by domchen** [comment, marked as answer]
Google actually uses a similar dependency synchronization strategy for their projects too. The depsync tool offers several key advantages compared to git submodules:
1. It is not only for checking out repositories, but also can be configured to download files from the Internet URL.
2. By default, the repositories are checked out with a depth of 1, ensuring that the main project maintains a relatively compact size at all times. While it's true that git submodule offers a similar capability, it requires explicitly specifying more intricate checkout arguments, making it a less straightforward approach.
3. You can configure separate repositories or internet files for various platforms. This way, you can avoid downloading unnecessary third-party dependencies and only acquire the ones specific to your current platform.

For more information, please check out the REAMED.md file in the [depsync](https://github.com/domchen/depsync) project.
